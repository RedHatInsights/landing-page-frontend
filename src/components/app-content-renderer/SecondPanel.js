import React from 'react';
import { Grid } from '@patternfly/react-core';
import recommendationRenderer from '../../components/app-content-renderer/recommendation-renderer';
import { useSelector } from 'react-redux';

const SecondPanel = () => {
  const recommendations = useSelector(
    ({ contentStore: { recommendations } }) => recommendations
  );

  return (
    <Grid className="pf-u-p-lg pf-u-pt-xl pf-u-pb-xl second-panel" hasGutter>
      {recommendationRenderer(recommendations)}
    </Grid>
  );
};

export default SecondPanel;
