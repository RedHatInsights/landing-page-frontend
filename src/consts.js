import React from 'react';
import Automation from './components/Automation.svg';
import Subscriptions from './components/Subscriptions.svg';
import Cost from './components/Cost.svg';
import Insights from './components/Insights.svg';
import Migrations from './components/Migrations.svg';
import Openshift from './components/Openshift.svg';

import insightsMarketing from './components/marketing/insightsMarketing.svg';
import openShiftMarketing from './components/marketing/openShiftMarketing.svg';
import ansibleMarketing from './components/marketing/ansibleMarketing.svg';

// isDevPreview for developer previews
// isPreview for technical previews
// isEarlyAccess for limited access early beta

export const activeTechnologies = [
    {
        entitlement: 'insights',
        marketing: true,
        marketingImage: insightsMarketing,
        marketingText: 'Proactively identify and remediate threats to security, performance, and stability.',
        marketingUrls: {
            learnMore: 'https://www.redhat.com/en/technologies/management/insights'
        },
        name: 'insights',
        id: 'insights',
        url: 'insights',
        baseApp: '/dashboard',
        apps: {
            dashboard: '/dashboard',
            patch: '/patch',
            advisor: '/advisor',
            drift: '/drift',
            vulnerability: '/vulnerability',
            policies: '/policies',
            compliance: '/compliance'
        },
        image: Insights,
        title: 'Red Hat Insights',
        emptyTitle: 'Red Hat Insights is included with every Red Hat Enterprise Linux subscription',
        emptyText: 'Proactively identify and remediate threats to security, performance, availability, \
        and stability with Red Hat Insights. Activate Red Hat Insights to get started today.',
        emptyID: 'insights',
        emptyAction: {
            primary: {
                title: 'Get started',
                navigate: 'https://www.redhat.com/wapps/eval/index.html?evaluation_id=1036'
            },
            close: {
                title: 'Not now'
            }
        },
        body: 'Identify and remediate configuration issues in your Red Hat® environments.'
    },
    {
        id: 'Openshift',
        url: 'openshift',
        baseApp: '/',
        apps: {
            'cluster manager': '/'
        },
        entitlement: 'openshift',
        marketing: true,
        marketingImage: openShiftMarketing,
        marketingText: 'Install, register, and manage Red Hat OpenShift® 4 clusters.',
        marketingUrls: {
            learnMore: 'https://try.openshift.com'
        },
        image: Openshift,
        emptyTitle: 'Get Started with Red Hat Openshift.',
        emptyID: 'openshift',
        emptyText: 'Test drive industry\'s leading container application platform in your browser, \
        and see how easy it is to use Kubernetes in your organization today.',
        emptyAction: {
            primary: {
                title: 'Try it free'
            }
        },
        title: 'Red Hat OpenShift Cluster Manager',
        body: 'Install, register, and manage Red Hat OpenShift® 4 clusters.'
    },
    {
        id: 'Ansible',
        entitlement: 'ansible',
        url: 'ansible',
        baseApp: '/automation-analytics',
        image: Automation,
        title: 'Red Hat Ansible Automation Platform',
        body: 'Extend your automation with analytics, policy and governance, and content management.',
        apps: {
            'automation analytics': '/automation-analytics',
            'automation hub': '/automation-hub',
            ...window.location.pathname.indexOf('/beta') === 0 && { 'automation services catalog': '/catalog' }
        },
        marketing: true,
        marketingImage: ansibleMarketing,
        marketingText: 'Extend your automation with analytics, policy and governance, and content management.',
        marketingUrls: {
            learnMore: 'http://ansible.com/products/automation-platform',
            tryIt: 'https://www.redhat.com/en/technologies/management/ansible/try-it'
        },
        emptyTitle: 'Ansible Automation Platform services requires a valid subscription',
        emptyText: 'Get analytics and knowledge on your automation, access to certified content, and more with a \
        Red Hat Ansible Automation Platform subscription.',
        emptyID: 'ansible',
        emptyAction: {
            // primary: {
            //     title: 'Request an evaluation',
            //     navigate: 'https://www.redhat.com/en/technologies/management/ansible/try-it'
            // },
            primary: {
                title: 'Learn more',
                navigate: 'https://www.ansible.com/products/automation-platform'
            },
            close: {
                title: 'Not now'
            }
        }
    },
    {
        id: 'Cost Management',
        url: 'cost-management',
        baseApp: '/',
        entitlement: 'cost_management',
        marketing: false,
        image: Cost,
        apps: {
            'cost management': '/'
        },
        emptyTitle: 'Cost Management for Red Hat OpenShift',
        emptyID: 'cost-management',
        emptyText: 'Cost Management provides visibility and analysis for your Red Hat OpenShift \
        and cloud costs. To obtain access to Cost Management, become a Red Hat OpenShift customer.',
        emptyAction: {
            primary: {
                title: 'Learn more',
                navigate: 'https://www.redhat.com/en/technologies/cloud-computing/openshift'
            },
            close: {
                title: 'Not now'
            }
        },
        title: 'Cost Management',
        body: 'Analyze, forecast and optimize your Red Hat OpenShift cluster costs in hybrid cloud environments.'
    },
    {
        id: 'migrations',
        entitlement: 'migrations',
        url: 'migrations',
        image: Migrations,
        title: 'Migration Services',
        body: 'Get recommendations on migrating your applications and infrastructure to Red Hat.',
        baseApp: '/migration-analytics',
        apps: {
            'migration analytics': '/migration-analytics'
        },
        marketing: false,
        emptyTitle: 'Migration Analytics requires a CloudForms subscription.',
        emptyID: 'migration-analytics',
        emptyText: 'Migration Analytics lets you examine workloads in your environment and evaluate \
        the effort needed to migrate or modernize each. Learn more to request a free CloudForms evaluation subscription',
        emptyAction: {
            primary: {
                title: 'Request an evaluation',
                navigate: 'https://access.redhat.com/products/red-hat-cloudforms-migrations/evaluation'
            },
            secondary: {
                title: 'Take a tour'
            },
            close: {
                title: 'Not now'
            }
        }
    },
    {
        id: 'subscriptions',
        entitlement: 'subscriptions',
        url: 'subscriptions',
        image: Subscriptions,
        title: 'Subscription Watch',
        body: 'Account-level summaries of your Red Hat subscription utilization',
        baseApp: '/rhel-sw',
        apps: {
            'Red Hat Enterprise Linux': '/rhel-sw',
            'Red Hat OpenShift': '/openshift-sw'
        },
        marketing: false,
        emptyTitle: 'Subscription Watch is an early access beta',
        emptyID: 'subscription-watch',
        emptyText: [
            'Subscription Watch helps you understand your total subscription usage and capacity over time. Take a quick tour to learn more.',
            <br key="sw1" />,
            <br key="sw2" />,
            'If you are interested in trying Subscription Watch, your Red Hat account team can help.'
        ],
        emptyAction: {
            primary: {
                title: 'Take a tour'
            },
            secondary: {
                title: 'Contact us',
                navigate: 'https://access.redhat.com/account-team'
            },
            close: {
                title: 'Not now'
            }
        }
    }
];
