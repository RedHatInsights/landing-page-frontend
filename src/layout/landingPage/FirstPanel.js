/* eslint-disable */
import React from 'react';
import FirstPanelTile from './FirstPanelTile';
import { 
  Title, 
  Flex, 
  FlexItem, 
  Gallery, 
  GalleryItem, 
  Stack, 
  StackItem 
} from '@patternfly/react-core'; 
import './styles/FirstPanel.scss';

const FirstPanel = () => {
  return (
    <>
      <Flex className="ins-p-console-landing-first-level">
        <Flex
          className="ins-c-landing-header__grouping"
          direction={{ default: 'column' }}
        >
          <FlexItem>
            <Title
              className="ins-c-landing-header__groupingTitle"
              headingLevel="h1"
              size="lg"
            >
              Application Services
            </Title>
          </FlexItem>
          <Flex>
            <FlexItem>
              <FirstPanelTile header="###" title="Kafka Clusters" />
            </FlexItem>
            <FlexItem>
              <FirstPanelTile header="###" title="API Objects" />
            </FlexItem>
            <FlexItem>
              <FirstPanelTile header="###" title="OpenDataHub Objects" />
            </FlexItem>
          </Flex>
        </Flex>
        <Flex
          className="ins-c-landing-header__grouping"
          direction={{ default: 'column' }}
        >
          <FlexItem>
            <Title className="ins-c-header--title" headingLevel="h1" size="lg">
              Platforms
            </Title>
          </FlexItem>
          <Flex className="ins-c-tile-grouping">
            <FlexItem>
              <FirstPanelTile header="###" title="API Objects" />
            </FlexItem>
            <FlexItem>
              <FirstPanelTile header="###" title="API Objects" />
            </FlexItem>
          </Flex>
        </Flex>
        <Flex
          className="ins-c-landing-header__grouping"
          direction={{ default: 'column' }}
        >
          <FlexItem>
            <Title className="ins-c-header--title" headingLevel="h1" size="lg">
              Automation
            </Title>
          </FlexItem>
          <Flex>
            <FlexItem>
              <FirstPanelTile header="###" title="API Objects" />
            </FlexItem>
          </Flex>
        </Flex>
      </Flex>
    </>
  );
};

export default FirstPanel;
