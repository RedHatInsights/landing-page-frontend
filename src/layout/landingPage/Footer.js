import React from 'react';
import {
  Flex,
  FlexItem,
  Split,
  SplitItem,
  Stack,
  StackItem,
  TextContent,
  Text,
  Title,
} from '@patternfly/react-core'; //eslint-disable-line
import {
  AutomationIcon,
  BlueprintIcon,
  CogIcon,
  DomainIcon,
  FilterIcon,
  IntegrationIcon,
} from '@patternfly/react-icons';
import './styles/ThirdPanel.scss';

const Footer = () => {
  return (
    <Flex className="ins-p-console-landing-third-level" hasGutter>
      <FlexItem>
        <Stack hasGutter>
          <StackItem>
            <Title headingLevel="h2" size="xl">
              Configure
            </Title>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <CogIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <FilterIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
        </Stack>
      </FlexItem>
      <FlexItem>
        <Stack hasGutter>
          <StackItem>
            <Title headingLevel="h2" size="xl">
              Try
            </Title>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <AutomationIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <IntegrationIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
        </Stack>
      </FlexItem>
      <FlexItem>
        <Stack hasGutter>
          <StackItem>
            <Title headingLevel="h2" size="xl">
              Learn
            </Title>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <BlueprintIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
          <StackItem>
            <Split>
              <SplitItem>
                <div className="footer-icon">
                  <DomainIcon />
                </div>
              </SplitItem>
              <SplitItem>
                <TextContent>
                  <Text component="h4">Lorem Dolor ipsum</Text>
                  <Text component="p">Lorem Dolor ipsum Lorem Dolor ipsum</Text>
                </TextContent>
              </SplitItem>
            </Split>
          </StackItem>
        </Stack>
      </FlexItem>
    </Flex>
  );
};

export default Footer;
