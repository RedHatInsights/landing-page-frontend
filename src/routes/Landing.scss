@import '~@redhat-cloud-services/frontend-components-utilities/styles/_mixins';
@import '~@patternfly/patternfly/sass-utilities/_all';

:root {
    --ins-global--Color--green: #37912f;
}

.ins-c-application-info .ins-m-gray { color: var(--pf-global--Color--200); }

.ins-m-tech-preview { color: var(--ins-global--Color--green); }

.ins-m-dev-preview { color: #39A5DC; }

.ins-m-early-preview {
    width: max-content;
    color: var(--pf-global--default-color--200);
}

.pf-c-modal-box__body .pf-c-empty-state img.ins-c-application-info__logo {
    @include rem('height', 76px);
}

.pf-c-card {

    &.ins-c-application-info {
        height: 100%;

        img.ins-c-application-info__logo {
            @include rem('height', 56px);
            max-width: initial;
        }

        svg.ins-c-application-info__logo {
            @include rem('height', 56px);
            @include rem('width', 56px);
        }

        h2 { margin-top: 15px; }
    }

    .ins-c-application-info__open {
        display: block;
        span { margin-right: 15px }
    }

    .ins-c-application-info__content-applist {
        display: flex;
        flex-direction: column;

        a {
            display: block;
            text-transform: capitalize;
        }

        a + a { margin-top: 10px; }
    }

    .ins-c-navigation { color: var(--pf-global--active-color--100); }
}

svg.ins-c-icon__active { fill: var(--pf-global--active-color--100); }

.pf-c-modal-box .pf-c-title { white-space: normal; }

sup.ins-c-rball { @include rem('font-size', 10px); }

.ins-c-application-info__split {
    height: 100%;
    .ins-c-application-info__open {
        margin-top: auto;
        margin-bottom: 0;
    }
}

// Unentitled Modal Styles
.ins-c-error-modal {
    &__header {
        text-align: center;
        max-width: 75%;
        margin-left: auto;
        margin-right: auto;
        padding-bottom: var(--pf-global--spacer--md);
    }
    &__content {
        align-items: center;
        justify-content: center;
        text-align: center;
        &--body { width: 75%; }
    }
}

// Alert message
.notifications-portal .pf-c-alert{
    .pf-c-alert__description { display: none; }
    .pf-c-alert__title {
        padding: unset;
    }

    .pf-c-alert__icon {
        padding: unset;
        background-color: unset;
    }
}

// Adjust PF layout to be able to display without header
.ins-c-page__landing-layout {
    height: 100%;

    .ins-c-page__landing-navigation {
        flex-shrink: 0;
    }
}

.ins-c-page__landing-content {
    width: 100%
}
