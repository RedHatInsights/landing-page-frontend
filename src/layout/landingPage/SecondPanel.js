import React from 'react';
import { Title, Flex, FlexItem } from '@patternfly/react-core';
import SecondPanelTile from './SecondPanelTile';
import './styles/SecondPanel.scss';

const SecondPanel = () => {
  return (
    <Flex className="ins-p-console-landing-second-level" hasGutter>
      <FlexItem>
        <Flex>
          <FlexItem className="ins-c-landing-grouping__title">
            <Title headingLevel="h1" size="xl">
              Application Services Recommendations
            </Title>
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
          <FlexItem className="ins-c-landing-grouping--separator">
            <SecondPanelTile />
          </FlexItem>
          <br />
          <FlexItem className="ins-c-landing-grouping__title">
            <Title headingLevel="h1" size="xl">
              Automation recommendations
            </Title>
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
        </Flex>
      </FlexItem>

      <FlexItem>
        <Flex>
          <FlexItem className="ins-c-landing-grouping__title">
            <Title headingLevel="h1" size="xl">
              Platform insights
            </Title>
          </FlexItem>
          <FlexItem>
            <SecondPanelTile title="Operational" />
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
          <FlexItem>
            <SecondPanelTile title="Financial" />
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
        </Flex>
      </FlexItem>

      <FlexItem>
        <Flex>
          <FlexItem className="ins-c-landing-grouping__title">
            <Title headingLevel="h1" size="xl">
              &nbsp;
            </Title>
          </FlexItem>
          <FlexItem>
            <SecondPanelTile title="Security" />
          </FlexItem>
          <FlexItem>
            <SecondPanelTile />
          </FlexItem>
        </Flex>
      </FlexItem>
    </Flex>
  );
};

export default SecondPanel;
