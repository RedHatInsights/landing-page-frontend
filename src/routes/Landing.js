import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';
import { withRouter } from 'react-router-dom';
import {
    Modal,
    Button,
    Title,
    Stack,
    StackItem
} from '@patternfly/react-core';

import Header from '../layout/Header';
import Body from '../layout/Body';
import Marketing from '../layout/Marketing';
import FooterTraditional from '../layout/FooterTraditional';
import { activeTechnologies } from '../consts';
import './Landing.scss';

class Landing extends Component {
    state = {
        isModalOpen: false
    }

    componentDidMount() {
        const { location } = this.props;

        const params = location.search.slice(1).split('&').reduce((acc, curr) => ({
            ...acc,
            [curr.split('=')[0]]: Object.values(activeTechnologies).find(item => item.entitlement === curr.split('=')[1])
        }), {});

        this.setState({
            ...params,
            isModalOpen: params && Object.keys(params).length > 0
        });

        window.insights.chrome.auth.getUser().then(user => {
            if (user) {
                this.setState({ unauthed: false });
            } else {
                this.setState({ unauthed: true });
            }
        }).catch(() => {
            this.setState({ unauthed: true });
        });

    }

    handleModalToggle = () => {
        this.setState({ isModalOpen: false });
    }

    render() {
        const { isModalOpen, not_entitled: notEntitled, unauthed } = this.state;

        return (
            <Fragment>
                { unauthed
                    ? <Marketing />
                    : <Fragment>
                        <Header />
                        <Body />
                    </Fragment>
                }
                <FooterTraditional />
                { notEntitled && <Modal
                    title={ 'You are not entitled to use this application' }
                    className='ins-c-error-modal'
                    hideTitle={ true }
                    isOpen={ isModalOpen }
                    onClose={ this.handleModalToggle }
                >
                    <Stack gutter='md' className='ins-c-error-state'>
                        <StackItem className='ins-c-error-state__title'>
                            <Title headingLevel="h3" size="2xl">{ notEntitled.emptyTitle }</Title>
                        </StackItem>
                        <StackItem className='ins-c-error-state__image'>
                            { notEntitled.icon && <notEntitled.icon
                                className="ins-c-icon__active"
                                aria-hidden
                                alt={ `${notEntitled.title} logo` }
                                { ...notEntitled.iconProps }
                            /> }
                            { notEntitled.image && <img
                                className="ins-c-application-info__logo"
                                aria-hidden
                                src={ notEntitled.image }
                                alt={ `${notEntitled.title} logo` } /> }
                        </StackItem>
                        <StackItem className='ins-c-error-state__body'>
                            { notEntitled.emptyText }
                        </StackItem>
                        <StackItem className='ins-c-error-state__footer'>
                            {
                                notEntitled.emptyAction.title &&
                                    <Button variant="primary" className='ins-c-error-state__footer-action' onClick={ () => {
                                        if (notEntitled.emptyAction.navigate) {
                                            window.location.href = notEntitled.emptyAction.navigate;
                                        }
                                    } } >
                                        { notEntitled.emptyAction.title }
                                    </Button>
                            }
                            <Button variant="link" className='ins-c-error-state__footer-close' onClick={ this.handleModalToggle }>
                                { notEntitled.emptyAction.close ? `${notEntitled.emptyAction.close }` : 'Close' }
                            </Button>
                        </StackItem>
                    </Stack>
                </Modal> }
            </Fragment>
        );
    }
}

Landing.propTypes = {
    history: PropTypes.object,
    loadTechnologies: PropTypes.func,
    location: PropTypes.shape({
        search: PropTypes.string
    })
};

Landing.defaultProps = {
    loadTechnologies: () => undefined
};

export default withRouter(Landing);
