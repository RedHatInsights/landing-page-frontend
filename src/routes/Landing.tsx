import React, { Fragment } from 'react';
import { useFlag } from '@unleash/proxy-client-react';

import '../components/app-content-renderer/styles/panels.scss';

import FirstPanel from '../components/app-content-renderer/first-panel';
import SecondPanel from '../components/app-content-renderer/second-panel';
import VirtualAssistant from '../components/app-content-renderer/virtual-assistant';
import { useLoadModule } from '@scalprum/react-core';
import { PageSection } from '@patternfly/react-core/dist/dynamic/components/Page';

const getWidgetLayoutLandingPage = () => {
  const scope = 'widgetLayout';
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: GridLayout }] = useLoadModule(
    { scope, module: './WidgetLayout' },
    {}
  );
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: AddWidgetDrawer }] = useLoadModule(
    { scope, module: './WidgetDrawer' },
    {}
  );
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: Header }] = useLoadModule(
    { scope, module: './WidgetHeader' },
    {}
  );
  if (!GridLayout || !AddWidgetDrawer || !Header) {
    return <></>;
  }
  return (
    <>
      <Header />
      <AddWidgetDrawer dismissible={false}>
        <PageSection>
          <GridLayout isLayoutLocked={false} />
        </PageSection>
      </AddWidgetDrawer>
    </>
  );
};

const Landing = () => {
  const widgetLayoutLandingPageEnabled = useFlag(
    'platform.landing-page.widgetization'
  );
  return (
    <div className="land-c-page-content pf-v5-u-display-flex pf-v5-u-flex-direction-column">
      <Fragment>
        <VirtualAssistant />
        {widgetLayoutLandingPageEnabled ? null : <FirstPanel />}
        {widgetLayoutLandingPageEnabled ? null : <SecondPanel />}
        {widgetLayoutLandingPageEnabled ? getWidgetLayoutLandingPage() : null}
      </Fragment>
    </div>
  );
};

export default Landing;
