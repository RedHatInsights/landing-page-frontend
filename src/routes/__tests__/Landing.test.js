/* eslint-disable react/display-name */
import React from 'react';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Modal, Title } from '@patternfly/react-core';

import * as notifications from '@redhat-cloud-services/frontend-components-notifications/redux/actions/notifications';

import Landing from '../Landing';

import * as actions from '../../store/actions';

import Footer from '../../components/app-content-renderer/footer';
import FirstPanel from '../../components/app-content-renderer/first-panel';
import SecondPanel from '../../components/app-content-renderer/second-panel';
import Loading from '../../layout/Loading';
import * as technologies from '../../consts';

jest.mock('../../components/app-content-renderer/first-panel', () => ({
  __esModule: true,
  default: () => <span>FirstPanel</span>,
}));

jest.mock('../../components/app-content-renderer/second-panel', () => ({
  __esModule: true,
  default: () => <span>SecondPanel</span>,
}));

jest.mock('../../components/app-content-renderer/footer', () => ({
  __esModule: true,
  default: () => <span>Footer</span>,
}));

jest.mock('../../consts', () => {
  const { activeTechnologies, ...rest } = jest.requireActual('../../consts');

  return {
    __esModule: true,
    ...rest,
    activeTechnologies: [
      {
        ...activeTechnologies[0],
        // disable image because img with svg has some issues
        image: undefined,
      },
      activeTechnologies.find(({ entitlement }) => entitlement === 'settings'),
      {
        ...activeTechnologies.find(
          ({ entitlement }) => entitlement === 'migrations'
        ),
        image: undefined,
      },
    ],
  };
});

const mockStore = configureMockStore();
const store = mockStore({});

describe('Landing component renders authenticated page', () => {
  let loadDataSpy;
  let addNotificationSpy;

  /**
   * Modal is appended to root not body
   */
  beforeAll(() => {
    const elem = document.createElement('div');
    elem.id = 'root';
    document.body.appendChild(elem);
  });

  afterAll(() => {
    const elem = document.getElementById('root');
    document.body.removeChild(elem);
  });

  beforeEach(() => {
    loadDataSpy = jest
      .spyOn(actions, 'loadData')
      .mockImplementation(() => ({ type: 'nonsense' }));
    addNotificationSpy = jest
      .spyOn(notifications, 'addNotification')
      .mockImplementation(() => ({ type: 'nonsense' }));
  });

  afterEach(() => {
    loadDataSpy.mockReset();
    addNotificationSpy.mockReset();
  });

  it('should render correctly', async () => {
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <Router>
            <Landing />
          </Router>
        </Provider>
      );
    });

    expect(wrapper.find(Loading)).toHaveLength(1);
    expect(wrapper.find(FirstPanel)).toHaveLength(0);
    expect(wrapper.find(SecondPanel)).toHaveLength(0);
    expect(wrapper.find(Footer)).toHaveLength(0);
    expect(wrapper.find(Modal)).toHaveLength(0);

    wrapper.update();

    expect(loadDataSpy).toHaveBeenCalled();
    expect(addNotificationSpy).not.toHaveBeenCalled();

    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(FirstPanel)).toHaveLength(1);
    expect(wrapper.find(SecondPanel)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1);
    // there is a modal in footer
    expect(wrapper.find(Modal)).toHaveLength(1);
  });

  it('isUnauthed', async () => {
    let wrapper;
    const spy = jest
      .spyOn(insights.chrome.auth, 'getUser')
      .mockImplementationOnce(() => Promise.resolve(undefined));

    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/']}>
            <Landing />
          </MemoryRouter>
        </Provider>
      );
    });

    wrapper.update();

    expect(loadDataSpy).toHaveBeenCalled();
    expect(addNotificationSpy).not.toHaveBeenCalled();

    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(FirstPanel)).toHaveLength(0);
    expect(wrapper.find(SecondPanel)).toHaveLength(0);
    expect(wrapper.find(Footer)).toHaveLength(0);

    spy.mockRestore();
  });

  it('isNotEntitled', async () => {
    let wrapper;
    const insightsInfo = technologies.activeTechnologies[0];

    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter
            initialEntries={[
              {
                pathname: '/beta',
                search: `not_entitled=${insightsInfo.entitlement}`,
              },
            ]}
          >
            <Landing />
          </MemoryRouter>
        </Provider>
      );
    });

    wrapper.update();

    expect(loadDataSpy).toHaveBeenCalled();
    expect(addNotificationSpy).not.toHaveBeenCalled();
    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(FirstPanel)).toHaveLength(1);
    expect(wrapper.find(SecondPanel)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1);
    expect(wrapper.find(Modal)).toHaveLength(2);

    let modalEntitled = wrapper.find(Modal).last();

    expect(modalEntitled.props().isOpen).toEqual(true);

    expect(modalEntitled.find(Title).text()).toEqual('');
    expect(
      modalEntitled.find('.ins-c-error-state__body').last().text()
    ).toEqual('');
    expect(
      modalEntitled.find('.ins-c-error-state__footer-action')
    ).toHaveLength(0);
    expect(
      modalEntitled.find('.ins-c-error-state__footer-secondary')
    ).toHaveLength(0);

    await act(async () => {
      wrapper.find('button.ins-c-error-state__footer-close').simulate('click');
    });

    wrapper.update();

    modalEntitled = wrapper.find(Modal).last();

    expect(modalEntitled.props().isOpen).toEqual(false);
  });

  it('isNotEntitled with alert', async () => {
    let wrapper;
    const settingsInfo = technologies.activeTechnologies[1];
    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter
            initialEntries={[
              {
                pathname: '/beta',
                search: `not_entitled=${settingsInfo.entitlement}`,
              },
            ]}
          >
            <Landing />
          </MemoryRouter>
        </Provider>
      );
    });

    wrapper.update();

    expect(loadDataSpy).toHaveBeenCalled();
    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(FirstPanel)).toHaveLength(1);
    expect(wrapper.find(SecondPanel)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1);
    expect(wrapper.find(Modal)).toHaveLength(1);

    expect(addNotificationSpy).toHaveBeenCalledWith({
      title: settingsInfo.emptyAlertTitle,
      variant: 'danger',
    });
  });

  it('isNotEntitled with secondary action', async () => {
    let wrapper;
    const migrationInfo = technologies.activeTechnologies[2];
    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter
            initialEntries={[
              {
                pathname: '/beta',
                search: `not_entitled=${migrationInfo.entitlement}`,
              },
            ]}
          >
            <Landing />
          </MemoryRouter>
        </Provider>
      );
    });

    wrapper.update();

    expect(loadDataSpy).toHaveBeenCalled();
    expect(addNotificationSpy).not.toHaveBeenCalled();
    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(FirstPanel)).toHaveLength(1);
    expect(wrapper.find(SecondPanel)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1);
    expect(wrapper.find(Modal)).toHaveLength(1);

    let modalEntitled = wrapper.find(Modal).last();

    expect(modalEntitled.props().isOpen).toEqual(false);
  });
});
