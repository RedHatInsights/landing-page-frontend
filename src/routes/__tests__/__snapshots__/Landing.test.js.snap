// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders authenticated page should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Landing)>
        <Route>
          <Landing
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            loadTechnologies={[Function]}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <Loading>
              <Connect(Main)>
                <Main>
                  <section
                    className="pf-l-page__main-section pf-c-page__main-section"
                    page-type=""
                  >
                    <Spinner
                      centered={true}
                    >
                      <div
                        className="ins-c-spinner ins-m-center"
                        role="status"
                      >
                        <span
                          className="pf-u-screen-reader"
                        >
                          Loading...
                        </span>
                      </div>
                    </Spinner>
                  </section>
                </Main>
              </Connect(Main)>
            </Loading>
          </Landing>
        </Route>
      </withRouter(Landing)>
    </Router>
  </BrowserRouter>
</Provider>
`;
